### DISCUSSION TITLE
Capture the response from the Location header

### DISCUSSION THREAD

**Discussion by JayasimhaNagaraj** [artilleryio/artillery#1088] [Artillery]
Hi Team,

For a particular request, i am getting a 302 response code and it has a code challenge in the Location header.
Could you please let us know how the **code_challenge** can be captured(**correlated**) to use in subsequent requests. 

Sample: 
http:response   "location":
https://<<>>/auth?client_id=<<>>&response_type=code&redirect_uri=<<>>&**code_challenge**=-fyF96Do9BeDsY34u51Axy86TYVBHUKyeCawGiG4L0E&code_challenge_method=S25

**Comment by hassy** [comment]
The best way to do that is with an `afterResponse` hook: https://artillery.io/docs/guides/guides/http-reference.html#afterResponse-hooks - the function would have access to the header, and the code_challenge can be extracted from it with a regular expression, and stored as a scenario variable, which can then be used in subsequent requests.
